controllers/argocd/statefulset.go

```diff
@@ -591,3 +596,24 @@ func updateNodePlacementStateful(existing *appsv1.StatefulSet, ss *appsv1.Statef
 		*changed = true
 	}
 }
+
+//to fetch the broken pod of StatefulSet if any, this function intends to delete the broken pod that cannot
+//be restarted due to known kubernetes issue https://github.com/kubernetes/kubernetes/issues/67250
+func getBrokenPod(ss *appsv1.StatefulSet, cr *argoprojv1a1.ArgoCD) *corev1.Pod {
+
+	k8s, err := initK8sClient()
+	if err != nil {
+		log.Error(err, "failed to initialize k8s client")
+	}
+
+	labelSel := common.ArgoCDKeyName + " = " + fmt.Sprintf("%s-%s", cr.Name, "application-controller")
+	po, err := k8s.CoreV1().Pods(ss.Namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: labelSel})
+	if err != nil {
+		log.Error(err, "failed to get Pod List")
```

Comment:
Should be: "Failed to list pods", and the error reason should be logged as well.

Reply:
I am logging out 'err', is there some other error reason that I should add?